lib/elasticity/base_document.rb

```diff
@@ -8,7 +8,7 @@ class BaseDocument
     # Configure the given klass, changing default parameters and resetting
     # some of the internal state.
     def self.configure(&block)
-      self.config = IndexConfig.new(Elasticity.config, &block)
+      self.config = IndexConfig.new(Elasticity.config, self.name.underscore.gsub("/", "_"), &block)
```

Comment:
I'd prefer keeping the `/` so that there are no conflicts and to keep the inverse mapping doable. Otherwise, both `Foo::Bar` and `FooBar` would both map to `foo_bar`.